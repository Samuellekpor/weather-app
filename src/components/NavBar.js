import React from 'react';
import { NavLink } from 'react-router-dom';
import left from '../images/icon-left.png';
import micro from '../images/microphone.png';
import setting from '../images/settings.png';

function NavBar() {
  return (
    <header className="nav-bar">
      <nav className="navigation">
        <NavLink to="/">
          <img src={left} className="icons" alt="back" />
        </NavLink>
        <div className="right-icons">
          <img src={micro} className="icons" alt="micro" />
          <img src={setting} className="icons" alt="settings" />
        </div>
      </nav>
    </header>
  );
}

export default NavBar;
