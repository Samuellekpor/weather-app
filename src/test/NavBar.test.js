import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import NavBar from '../components/NavBar';

it('NavBar component should render correctly', () => {
  const nav = render(
    <Router>
      <NavBar />
    </Router>,
  );
  expect(nav).toMatchSnapshot();
});
