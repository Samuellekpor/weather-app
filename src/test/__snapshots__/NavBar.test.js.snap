// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="nav-bar"
      >
        <nav
          class="navigation"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt="back"
              class="icons"
              src="icon-left.png"
            />
          </a>
          <div
            class="right-icons"
          >
            <img
              alt="micro"
              class="icons"
              src="microphone.png"
            />
            <img
              alt="settings"
              class="icons"
              src="settings.png"
            />
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="nav-bar"
    >
      <nav
        class="navigation"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <img
            alt="back"
            class="icons"
            src="icon-left.png"
          />
        </a>
        <div
          class="right-icons"
        >
          <img
            alt="micro"
            class="icons"
            src="microphone.png"
          />
          <img
            alt="settings"
            class="icons"
            src="settings.png"
          />
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
